### PR TITLE
crush the vectorized method resistance

### LINKED ISSUE #22959: some functions are still implicitly vectorized for SparseVector
I haven't seen an issue or PR for this, but feel free to close if I'm missing something.

base/sparse/sparsevector.jl still has method definitions for sin, tan, exp, etc. that operate element-wise over SparseVectors.

### PR DESCRIPTION
**Pull request by Sacha0**
Fly the red flag and grant them no quarter.

Fixes #22959. Needs news? Best!

### CODE REVIEW

File: base/sparse/sparsevector.jl [hunk 1]
```diff
@@ -1094,16 +1083,6 @@ macro unarymap_z2nz(op, TF)
     end)
 end
 
-for op in [:exp, :exp2, :exp10, :log, :log2, :log10,
-           :cos, :csc, :cot, :sec, :cospi,
-           :cosd, :cscd, :cotd, :secd,
-           :acos, :acot, :acosd, :acotd,
-           :cosh, :csch, :coth, :sech,
-           :acsch, :asech]
-    @eval @unarymap_z2nz $(op) Number
```
Comment (tkelman) [inline base/sparse/sparsevector.jl]: we could also deprecate this macro? (or remove if not used outside of base)

Comment (Sacha0) [inline base/sparse/sparsevector.jl]: Absolutely, chances are most of the `Xmap_YZ` macros/functions for `SparseVector`s can be cleaned out with some care :). Would a separate pull request be alright?

**Comment by martinholters** [comment]
https://www.youtube.com/watch?v=8CP9dg38cAI

**Comment by fredrikekre** [comment]
Perhaps add a NEWS bullet and this should be good to go?

**Comment by Sacha0** [comment]
> Perhaps add a NEWS bullet and this should be good to go?

Thanks for the bump! Added a commit with news for this pull request and updated news PR links. Best!

**Review by fredrikekre — APPROVED** [review]
Travis is one timeout and Arnoldi.

**Comment by Sacha0** [comment]
Thanks all! :)

**Comment by JeffBezanson** [comment]
This is truly a glorious moment.

**Comment by StefanKarpinski** [comment]
Wow, is this really the last of the vectorized methods? That is momentous.

**Comment by Sacha0** [comment]
I suspect pockets of resistance remain, hopefully to be smoked out prior to The Great Feature Freeze.

**Comment by StefanKarpinski** [comment]
Hmm. I wonder if there's a way to automatically test for vectorized methods...

**Comment by StefanKarpinski** [comment]
Ok, I found some lingerers by the very high tech means of grepping for `vectoriz` in the code base:

https://github.com/JuliaLang/julia/blob/0f4229d2e744375255fa1b27cb180f678ebf3e76/base/dates/io.jl#L561-L580

**Pull request merged**